.circleci/scripts/should_run_job.sh

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+set -ex -o pipefail
+
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+# Check if we should actually run
+echo "BUILD_ENVIRONMENT: ${BUILD_ENVIRONMENT}"
+echo "CIRCLE_PULL_REQUEST: ${CIRCLE_PULL_REQUEST:-}"
+if ! [ -z "${CIRCLE_PULL_REQUEST:-}" ]; then
```

Comment:
Do you have a reason for ``` if ! [ -z "..." ]; then``` vs ```if [[ -n "..." ]]; then``` ?

Reply:
The proximal reason is that the old code did it this way, and I just copy pasted it. The original reason is probably, I could remember what `-z` was, I couldn't remember what the inverse was, so I just negated it.